Improves logic when checking for a valid file and directory names.

qchaupds opened the pull request:
1. Add github5 test artifacts to check for bad file and directory names
2. Refactor, improve logic to check for invalid filename, and make static variable private in FileAndDirectoryNamingRule.java
3. Use refactored functions in FileAndDirectoryNamingChecker.java when validating a stand alone label with the -R pds.label parameter
4. Add github5 test to ValidationIntegrationTests.java

Refs #5 https://github.com/nasa-pds/validate/issues/5 Improve file base name check according to Standards Reference

Closes NASA-PDS/validate#5

The issue is the code was checking for the name without the extension, thus would flag 'core.vmac' was invalid.  Improvements have
been made to check the name of the file plus extension so 'core.vmac' would be valid but 'core' would be invalid.

Tested in DEV:

% validate -R pds4.bundle -r report_github5_bundle_invalid.json -s json -t src/test/resources/github5/invalid/bundle_kaguya_derived.xml

% validate -R pds4.label -r report_github5_label_invalid.json -s json -t src/test/resources/github5/invalid/VALID_odf07155_msgr_11_with_bad_char_\#_.xml

Bonus tests (good bundle and valid label):

% validate -R pds4.bundle -r report_github5_bundle_valid.json -s json -t src/test/resources/github5/valid/bundle_kaguya_derived.xml

% validate -R pds4.label --skip-content-validation -r report_github5_label_valid.json -s json -t src/test/resources/github5/valid/VALID_odf07155_msgr_11.xml

The first set of 2 tests exercise the logic to catch a bundle/label with bad file names and bad directory name.

The following labels should FAIL due to new logic:

- src/test/resources/github5/invalid/3juno_lwr01896_ines%20fits%20headers.pdfa.xml,
- src/test/resources/github5/invalid/VALID_odf07155_msgr_11_with_bad_char_%23_.xml,
- src/test/resources/github5/invalid/a.out,
- src/test/resources/github5/invalid/lpt8,
- src/test/resources/github5/invalid/com4,
- src/test/resources/github5/invalid/core,
- src/test/resources/github5/invalid/file_with_bad_char_%23_.txt,
- src/test/resources/github5/invalid/3juno_lwr01896_ines%20fits%20headers.pdfa.xml,
- src/test/resources/github5/invalid/VALID_odf07155_msgr_11_with_bad_char_%23_.xml,
- src/test/resources/github5/invalid/my_dir.ext/

The following previously suspected files should PASS now:

- src/test/resources/github5/invalid/core.vmac,
- src/test/resources/github5/invalid/file_aux_should_be_ok.tab,
- src/test/resources/github5/invalid/lpt9.tab,
- src/test/resources/github5/invalid/lpt9_should_be_ok.tab,
- src/test/resources/github5/invalid/this_file_should_be_ok_core.tab

The second set of tests are to make sure the code still run for valid a bundle and a valid label.

Commit:
Improves logic when checking for a valid file and directory names.

1. Add github5 test artifacts to check for bad file and directory names
2. Refactor, improve logic to check for invalid filename, and make static variable private in FileAndDirectoryNamingRule.java
3. Use refactored functions in FileAndDirectoryNamingChecker.java when validating a stand alone label with the -R pds.label parameter
4. Add github5 test to ValidationIntegrationTests.java

Refs #5 https://github.com/nasa-pds/validate/issues/5 Improve file base name check according to Standards Reference

jordanpadams reviewed (changes requested):
@qchaupds looks like the integration tests are failing.

additionally, it looks like it does not perform this check when you try to validate an individual file. if update `src/test/resources/github5/invalid/data_spectra/kgrs_calibrated_spectra_per1.xml` to point to a knew file i created `<file_name>lpt1.foo</file_name>`, it passes successfully. i tried the same thing for a file names `core` with the same passing result.

```
$ validate-1.25.0-SNAPSHOT/bin/validate -t src/test/resources/github5/invalid/data_spectra/kgrs_calibrated_spectra_per1.xml --skip-content-validation

PDS Validate Tool Report

Configuration:
   Version                       1.25.0-SNAPSHOT
   Date                          2021-01-08T20:16:16Z

Parameters:
   Targets                       [file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github5/invalid/data_spectra/kgrs_calibrated_spectra_per1.xml]
   Severity Level                WARNING
   Recurse Directories           true
   File Filters Used             [*.xml, *.XML]
   Data Content Validation       off
   Product Level Validation      on
   Max Errors                    100000
   Registered Contexts File      /Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/validate-1.25.0-SNAPSHOT/resources/registered_context_products.json



Product Level Validation Results

  PASS: file:/Users/jpadams/Documents/proj/pds/pdsen/workspace/validate/src/test/resources/github5/invalid/data_spectra/kgrs_calibrated_spectra_per1.xml
        1 product validation(s) completed

Summary:

  0 error(s)
  0 warning(s)

  Product Validation Summary:
    1          product(s) passed
    0          product(s) failed
    0          product(s) skipped

  Referential Integrity Check Summary:
    0          check(s) passed
    0          check(s) failed
    0          check(s) skipped


End of Report
Completed execution in 4694 ms
```

Commit:
Fix integration tests issues and add logic to handle -R pds.label when file_name tag contains invalid names.

1. Add class FileAreaExtractor.java to handle -R pds.label when file_name tag contains invalid names
2. Add logic to call functions in FileAreaExtractor to check for invalid names in LabelValidationRule.java
3. Improve error messages to alert if the label has failed to parse in LabelValidationRule.java
4. Improve error messages to print the expected and received count of error messages in ValidationIntegrationTests.java

Refs #5 https://github.com/nasa-pds/validate/issues/5 Improve file base name check according to Standards Reference
